scipy/stats/_binned_statistic.py

```diff
@@ -15,7 +15,7 @@
 
 
 def binned_statistic(x, values, statistic='mean',
-                     bins=10, range=None):
+                     bins=10, _range=None):
```

Comment:
This breaks backwards compatibility since parameter names are effectively public API. Can you please undo this change, here and elsewhere.

Reply:
Unfortunately __range__ is a Python builtin in both Python 2 and Python 3.  That makes it a poor choice for a variable name.

Reply:
Indeed. But that ship has likely sailed. 

This is not to say we never rename poorly named arguments. We can, if the benefit outweights a possible disruption downstream. Which I don't think it does, in this particular case.

Reply:
What do you want to call the builtin __range()__??
1. xrange()?
2. builtin_range()?
3. range_builtin()?
4. range_function()?
5. other?

Reply:
In this function we don't call the builtin `range`, so there's no conflict and nothing needs to be changed.

Reply:
The problem comes in __binned_statistic_dd()__ and ___bin_edges()__ that use both the builtin __range()__ and the function parameter __range__.

Reply:
How about an explicit `import builins; builtins.range(...)`?

Reply:
Nice!  (For a separate PR.)